server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/redirect/ProcessRedirectNotify.java

```diff
@@ -86,7 +87,9 @@ public void process(Mail originalMail) throws MessagingException {
 
             if (senderDomainIsValid(newMail)) {
                 // Send it off...
-                mailet.getMailetContext().sendMail(newMail);
+                if (!newMail.getRecipients().isEmpty()) {
+                    mailet.getMailetContext().sendMail(newMail);
```

Comment:
why it's not handled by `sendMail` ? I mean, is there any legitimate use of `sendMail` without any recipient ?

Reply:
I did have the same thought than you but I did not want to put too much logic in `FakeMailContext`

After thinking about it, *I considered preventing bad behaviours in the first place was desirable*